fr/lessons/basics/documentation.md

```diff
@@ -0,0 +1,340 @@
+
+Exemples
+
+    iex> Greeter.hello("Sean")
+    "Hello, Sean"
+
+    iex> Greeter.hello("pete")
+    "Hello, pete"
+
+iex>
+```
+
+Vous remarquerez que vous pouvez utiliser une syntaxe d'embelissement dans la documentation, et que votre terminal en fera le rendu.
+Mis à part le fait que cette fonctionnalité vienne s'ajouter à un déjà vaste écosystème, elle devient beaucoup plus intéressante quand on en vient à
+ExDoc pour générer de la documentation HTML à la volée.
+
+## ExDoc
+
+ExDoc est un projet officiel d'Elixir qui **produit de la documentation HTML pour les projets Elixir**, et peut être trouvé sur [GitHub](https://github.com/elixir-lang/ex_doc).
+Créons d'abord notre premier projet en Elixir :
+
+```bash
+$ mix new greet_everyone
+
+* creating README.md
+* creating .gitignore
+* creating mix.exs
+* creating config
+* creating config/config.exs
+* creating lib
+* creating lib/greet_everyone.ex
+* creating test
+* creating test/test_helper.exs
+* creating test/greet_everyone_test.exs
+
+Your Mix project was created successfully.
+You can use "mix" to compile it, test it, and more:
+
+    cd greet_everyone
+    mix test
+
+Run "mix help" for more commands.
+
+$ cd greet_everyone
+```
+
+Copiez et collez le code de la leçon sur l'annotateur `@doc` dans un fichier appelé `lib/greeter.ex` et assurez-vous que tout compile bien.
+Maintenant que vous travaillez dans un projet Mix, nous avons besoin de démarrer IEx un peu différemment, en utilisant la commande `iex -S mix` : 
+
+```bash
+iex> h Greeter.hello
+
+                def hello(nom)
+
+`hello/1` Affiche un message de salutation
+
+Paramètres
+
+    • nom : Une chaîne de caractères (String) qui représente le nom de la personne.
+
+Exemples
+
+    iex> Greeter.hello("Sean")
+    "Hello, Sean"
+
+    iex> Greeter.hello("pete")
+    "Hello, pete"
+
+```
+
+### Installer
+
+Si tout va bien et que nous voyons l'affichage ci-dessus, c'est que nous sommes prêts pour installer ExDoc. Dans notre fichier `mix.exs`, ajoutons les deux 
+dépendances nécessaires : `earmark` et `:ex_doc`.
+
+```elixir
+  def deps do
+    [{:earmark, "~> 0.1", only: :dev},
+    {:ex_doc, "~> 0.11", only: :dev}]
+  end
+```
+
+On spécifiera l'option `only: :dev` car nous n'avons pas besoin de ces dépendances dans un environnement de production. Mais pourquoi Earmark ? Et bien, c'est un 
+analyseur syntaxique, ou _parser_, de Markdown pour Elixir et utilisé par ExDoc pour produire de magnifiques fichiers HTML à partir de la documentation au sein 
+de `@moduledoc` et `@doc`.
+
+Gardez à l'esprit que vous n'êtes pas obligés d'utiliser Earmark. Vous pouvez très facilement vous tourner vers d'autres outils tels que 
+Pandoc, Hoedown ou Cmark. Néanmoins, vous aurez besoin d'un peu plus de configuration, que vous pourrez trouver
+[à cette addresse](https://github.com/elixir-lang/ex_doc#changing-the-markdown-tool)
+
+### Générer la documentation
```

Comment:
It's necessary to add a custom anchor here too, because Jekyll will skip the accented letters and name that header as `gnrer-la-documentation` and the summary's anchor will not work.

If you change it for
```
### <a name="generer-la-documentation"></a>Générer la documentation
```
The summary will work properly